### DISCUSSION TITLE
[UI/UX] Improved modals

### DISCUSSION THREAD

**Discussion by dstrukt** [btcpayserver/btcpayserver#2589] [General]
We currently have a pattern of showing modals that do not sit above the corresponding view, fortunately the modals, and this pattern is consistent.

See:

![Screen Shot 2020-12-14 at 1 52 1](https://user-images.githubusercontent.com/6250771/122685217-d0159980-d1be-11eb-82fe-d7669224416a.png)

However, I'd like to propose that we have modals sit above the content they correspond to.

For example:

<img width="1012" alt="Screen Shot 2021-06-20 at 11 57 40 AM" src="https://user-images.githubusercontent.com/6250771/122685208-cb50e580-d1be-11eb-83bb-a690091c2ec7.png">

I could provide more examples, and will if necessary, but you get the idea.

This would be audited and applied across the app. The modals themselves, other than potentially a little wording and aesthetic clean up, would not change much.

I'm not sure how much work this would require, but @dennisreimann and I have discussed it, and we might step through 1 example together, and if possible, I could try to help.

Feedback & thoughts welcome!

**Comment by dennisreimann** [comment]
I can envision this as a good JavaScript-enabled alternative to what we currently have. Ideally we can do it in a way, that reuses the same partial for the JS modal and the non-JS fallback (separate page as currently).

**Reply by dstrukt** [reply]
Sounds good, feel free to take the lead, or if it's feasible, we could work on this together in a pair dev. session.

**Reply by dennisreimann** [reply]
Added #2614 to advance the discussion with something tangible. :)

**Reply by bolatovumar** [reply]
Yeah, I've been thinking about this for a while. Would be nice to have JS-powered modals if JS is enabled and the separate page if it's not.

**Reply by dennisreimann** [reply]
Yes, that's how we'll do it.

**Comment by pavlenex** [comment]
Concept ACK.

**Comment by Zaxounette** [comment]
ACK, as discussed in the [Design Call #11](https://github.com/btcpayserver/organization/issues/40).

**Comment by NicolasDorier** [comment]
Concept ACK
